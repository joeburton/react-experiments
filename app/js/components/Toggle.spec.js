import React from 'react';
import { Toggle } from './Toggle';
import renderer from 'react-test-renderer';

test('Toggle ON OFF', () => {
    const component = renderer.create(
        <Toggle/>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    tree.props.onClick();
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    tree.props.onClick();
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});