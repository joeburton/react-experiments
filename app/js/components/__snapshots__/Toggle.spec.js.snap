// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle ON OFF 1`] = `
<button
  onClick={[Function]}
>
  ON
</button>
`;

exports[`Toggle ON OFF 2`] = `
<button
  onClick={[Function]}
>
  OFF
</button>
`;

exports[`Toggle ON OFF 3`] = `
<button
  onClick={[Function]}
>
  ON
</button>
`;
